app/code/Ometria/Api/Controller/V1/Attribute/Types.php

```diff
@@ -1,58 +1,101 @@
 <?php
 namespace Ometria\Api\Controller\V1\Attribute;
+
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Catalog\Model\ResourceModel\Product\Attribute\CollectionFactory as AttributeCollectionFactory;
+use Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection as AttributeCollection;
+use Magento\Catalog\Api\Data\ProductAttributeInterface;
 use Ometria\Api\Helper\Format\V1\Attribute\Types as Helper;
-use \Ometria\Api\Controller\V1\Base;
+use Ometria\Api\Controller\V1\Base;
+
 class Types extends Base
 {
     protected $resultJsonFactory;
-    protected $attributes;
-
-	public function __construct(
-		\Magento\Framework\App\Action\Context $context,
-		\Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
-		\Magento\Catalog\Model\ResourceModel\Product\Attribute\Collection $attributes		
-	) {
-		parent::__construct($context);
-		$this->resultJsonFactory = $resultJsonFactory;
-		$this->attributes = $attributes;		
-	}
-
-    protected function serializeAttribute($attribute)
+    protected $attributeCollectionFactory;
+
+    public function __construct(
+        Context $context,
+        JsonFactory $resultJsonFactory,
+        AttributeCollectionFactory $attributeCollectionFactory
```

Comment:
This is a tidy up of some legacy code. Using the factory here ensures the class has a 'newable' collection rather than a singleton, which aligns with the core Magento code standard. Note a new instance of the collection is created from the factory on line 29.